src/components/City.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { API_BASE_URL, API_KEY } from '../api';
+import { setCity } from '../redux/city/citySlice';
+
+const City = () => {
+  const dispatch = useDispatch();
+  const city = useSelector((state) => state.city);
+  const { state: selectedState, city: selectedCity } = useParams();
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const response = await fetch(`${API_BASE_URL}/city?city=${selectedCity}&state=${selectedState}&country=Canada&key=${API_KEY}`);
+      if (!response.ok) {
+        throw new Error('Failed to fetch data');
+      }
+      const data = await response.json();
+      dispatch(setCity(data.data));
+      setLoading(false);
+    } catch (error) {
+      setError(error);
+      setLoading(false);
+    }
+  }, [dispatch, selectedState, selectedCity]);
+
+  useEffect(() => {
+    fetchData();
+  }, [fetchData, selectedState, selectedCity]);
+
+  const retryFetch = () => {
+    setError(null);
+    setLoading(true);
+    fetchData();
+  };
+
+  const renderContent = () => {
+    if (loading) {
+      return <p className="loading">Loading...</p>;
+    }
+    if (error) {
+      return (
+        <div className="error">
+          <p>Error fetching data. Please try again.</p>
+          <button type="button" onClick={retryFetch}>Retry</button>
+        </div>
+      );
+    }
+    const { current } = city;
+    return (
+      <div className="container">
+        <nav className="navbar flex">
+          <Link to={`/${selectedState}`} className="back">Back</Link>
+          <h2>{`${selectedCity} weather and air pollution details`}</h2>
+        </nav>
+        <article className="details">
+          <h3>Pollution Details</h3>
+          <p>
+            Air Quality Index (AQI):
+            {current.pollution.aqius}
+          </p>
+          <p>
+            Main Pollutant:
+            {current.pollution.mainus}
+          </p>
+          <h3>Weather Details</h3>
+          <p>
+            Temperature:
+            {current.weather.tp}
+            °C
+          </p>
+          <p>
+            Pressure:
+            {current.weather.pr}
+            {' '}
+            hPa
+          </p>
+          <p>
+            Humidity:
+            {current.weather.hu}
+            %
+          </p>
+        </article>
+      </div>
+    );
```

Comment:
- [ ] As required for the project to follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including `Typography`(font face, size, and weight), `Layout`(composition and space between elements), I think it would be better if you improve the alignment of your element text, rows to have some different colors(light and dark).

| Template Design | Your Design |
| ---------------------- | ------------------- |
| ![Screenshot from 2023-09-14 04-10-01](https://github.com/tajemouti/weather-pollution-metrics/assets/83042742/edaf2194-5052-4fda-b065-a6348a51e9b1) | ![Screenshot from 2023-09-14 04-11-15](https://github.com/tajemouti/weather-pollution-metrics/assets/83042742/b22e4cf9-e81b-4827-a350-bdf2f1ec47cc) |